api/repo/repair.go

```diff
@@ -116,6 +116,35 @@ func RepairRepo(c *gin.Context) {
 		}
 	}
 
+	// get repo information from the source
+	sourceRepo, err := scm.FromContext(c).GetRepo(ctx, u, r)
+	if err != nil {
+		retErr := fmt.Errorf("unable to retrieve repo info for %s from source: %w", sourceRepo.GetFullName(), err)
+
+		util.HandleError(c, http.StatusBadRequest, retErr)
+
+		return
+	}
+	// if repo has a name change, then update DB with new name
+	// if repo has a org change, update org as well
+	if sourceRepo.GetName() != r.GetName() || sourceRepo.GetOrg() != r.GetOrg() {
+		r.SetPreviousName(r.GetName())
```

Comment:
not sure that's it's crucial, but do we only want to do this if repo name specifically changed? as it stands, it will overwrite previous name if only org changed?